addons/website/tests/test_get_current_website.py

```diff
@@ -134,3 +137,36 @@ def test_02_signup_user_website_id(self):
 
         user = self.env['res.users'].create({'website_id': website.id, 'login': 'sad@mail.com', 'name': 'Hope Fully'})
         self.assertTrue(user.website_id == user.partner_id.website_id == website)
+
+@tagged('post_install', '-at_install')
+class TestRecursionIssue(HttpCase):
+    def test_recursive_current_website(self):
+        Rule = self.env['ir.rule']
+        Rule.create({
+            'name': 'Recursion',
+            'model_id': self.env.ref('website.model_website').id,
+            'domain_force': [(1, '=', 1)],
+            'groups': [],
+        })
+
+        website = self.env.ref('website.default_website')
+        base_url = website.get_base_url()
+        login_form_request = self.url_open(base_url + '/web/login')
+        content = login_form_request.content.decode("utf8")
+        el = html.fromstring(content)
+        csrf_input = el.xpath("//form[@class='oe_login_form']//input[@name='csrf_token']")[0]
+        csrf = csrf_input.value
+        login_request = requests.post(base_url + "/web/login", {
```

Comment:
@odoo/rd-security This is a test that tries to replicate an actual login post request as if it was sent from the form.